### ISSUE TITLE
Split Sections to Files

### ISSUE THREAD

**Issue by EmpireWorld** [dexteryy/spellbook-of-modern-webdev#6]
Hi.

It is better to split sections to separated files for more readability and ease of use.

For example split section Node.js to a node-js.md file.

Thanks.

Best Regards.

**Comment by dexteryy** [comment]
@EmpireWorld Thanks for the suggestion. It is a real problem that should be solved. However, multi-page repo has its drawbacks, such as not being able to use browser's find feature, not being able to see the whole thing and scroll through all of it (like a mindmap).

I've got a plan to build a web app which can be automatically generated from the text pattern and structure in the current single-file document. A web app can be much more efficient and can support some useful features.

**Comment by EmpireWorld** [comment]
You can provide a **table of contents** file by links to the corresponding files. that supports browser search.

But the Web App idea is better.

**Comment by Albwrt** [comment]
Think you For that

**Comment by marlazinger200** [comment]
1

**Comment by tythewebguy** [comment]
For what it's worth, I prefer having a single file. 

Perhaps we can identify the problem being solved by splitting things into multiple files. There may be other solutions available.

**Comment by EmpireWorld** [comment]
He can provide both of them, first he made a single file, then he make multiple files. but it makes work harder for himself, because when he wants to edit the single file he should scrimmage with the big data. but when there were split content there is no need to scrimmage.

**Comment by tythewebguy** [comment]
Yes, but what exactly is the problem being solved? Is the problem readability and ease of use? If so, what makes it difficult to read and use? Jumping between TOC and content? The length of the content?

I'm just trying to understand the problem before jumping to a solution. There may be other solutions that make it easier to manage and use the content. Just trying to take a step back and see the big picture instead of zooming in on a specific solution.

**Comment by EmpireWorld** [comment]
You are smart man, you are right, but the i am thinking about everyone behaviour.

Think about a mobile agent or tablet, it makes it hard for the user to find the wanted content.

Re: Yes, i mean the readability and ease of use for both reader and writer. 

Re: The length of the content. Makes it hard for reader and writer. writer should search for the content in a long way but when he can it in a separated file.

If you have any solutions we are here to listen.

It is just a recommendation.

Thanks.

**Comment by chuanqisun** [comment]
Agree with you @dexteryy, reader experience and author experience are two somewhat conflicting forces. File splitting might help the author organize content but may undermine the search/find experience for the reader. Another draw back is that once you split the files, inter-linking between files will become fragile.

**Comment by ycaptain** [comment]
> @EmpireWorld Thanks for the suggestion. It is a real problem that should be solved. However, a multi-page repo has its drawbacks, such as not being able to use browser's find feature, not being able to see the whole thing and scroll through all of it (like a mindmap).
> 
> I've got a plan to build a web app which can be automatically generated from the text pattern and structure in the current single-file document. A web app can be much more efficient and can support some useful features.

Any progress? Maybe you can create a repo to start the web app and we could try to contribute to it.
